src/JBOptimismSucker.sol

```diff
@@ -1,56 +1,70 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.23;
 
-import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
-import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import {IJBPrices} from "@bananapus/core/src/interfaces/IJBPrices.sol";
-import {IJBRulesets} from "@bananapus/core/src/interfaces/IJBRulesets.sol";
 import {IJBDirectory} from "@bananapus/core/src/interfaces/IJBDirectory.sol";
-import {IJBTokens} from "@bananapus/core/src/interfaces/IJBTokens.sol";
 import {IJBPermissions} from "@bananapus/core/src/interfaces/IJBPermissions.sol";
+import {IJBTokens} from "@bananapus/core/src/interfaces/IJBTokens.sol";
 import {JBConstants} from "@bananapus/core/src/libraries/JBConstants.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
 
-import {JBSucker, IJBSuckerDeployer, JBAddToBalanceMode} from "./JBSucker.sol";
+import {JBSucker} from "./JBSucker.sol";
+import {JBOptimismSuckerDeployer} from "./deployers/JBOptimismSuckerDeployer.sol";
+import {IJBOptimismSucker} from "./interfaces/IJBOptimismSucker.sol";
+import {IJBSuckerDeployer} from "./interfaces/IJBSuckerDeployer.sol";
+import {IOPMessenger} from "./interfaces/IOPMessenger.sol";
+import {IOPStandardBridge} from "./interfaces/IOPStandardBridge.sol";
+import {JBAddToBalanceMode} from "./enums/JBAddToBalanceMode.sol";
+import {JBInboxTreeRoot} from "./structs/JBInboxTreeRoot.sol";
 import {JBMessageRoot} from "./structs/JBMessageRoot.sol";
+import {JBOutboxTree} from "./structs/JBOutboxTree.sol";
 import {JBRemoteToken} from "./structs/JBRemoteToken.sol";
-import {JBInboxTreeRoot} from "./structs/JBInboxTreeRoot.sol";
-import {JBOptimismSuckerDeployer} from "./deployers/JBOptimismSuckerDeployer.sol";
-import {OPMessenger} from "./interfaces/OPMessenger.sol";
-import {OPStandardBridge} from "./interfaces/OPStandardBridge.sol";
 import {MerkleLib} from "./utils/MerkleLib.sol";
 
 /// @notice A `JBSucker` implementation to suck tokens between two chains connected by an OP Bridge.
-contract JBOptimismSucker is JBSucker {
-    using MerkleLib for MerkleLib.Tree;
+contract JBOptimismSucker is JBSucker, IJBOptimismSucker {
     using BitMaps for BitMaps.BitMap;
+    using MerkleLib for MerkleLib.Tree;
 
-    event SuckingToRemote(address token, uint64 nonce);
+    //*********************************************************************//
+    // --------------------------- custom errors ------------------------- //
+    //*********************************************************************//
+
+    error JBOptimismSucker_TokenNotMapped();
+    error JBOptimismSucker_UnexpectedMsgValue();
```

Comment:
Why are we not using `JBSucker` errors? If we change the errors each sucker then a client needs to be aware of all sucker implementations to display errors properly.